Feat: Clockwork mender (balance and fixes)

Den109G opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->
## What Does This PR Do

Меняет баланс по [заметкам Биба](https://discord.com/channels/617003227182792704/981565006762106891/997387930358255716):
- Меняет время для зачарований (телепортация, сокрытие шестерней, ускорения и другие).
- Даёт защиту от ЭМИ киборгам после Ratvar_act.
- Меняет мультитул у cogscarab (чисто декор).
- Увеличение урона у копья, меча.
- Уменьшение урона у щита.
- Увеличивает скорость перемещения у зачарованной робы.

Меняет чуть-чуть код вещей, чтобы балансерам легче было менять циферки (у кирасы - absorb_uses и reflect_uses возвращаются через initial(), то же самое и с мечом и его уроном).

Чинит баги:
- Копьё клокеров; теперь им можно ударить (спасибо Бибу).
- Заклинание Harden plates корректно работает (спасибо Бибу).

Добавляет:
- Киборги собранные через Soul Vessel получают способности клокеров.
- Шесткарабеи получают язык дронов (который `:d` ).
- Обычный экипаж не может пользоваться часами с телепортом/сокрытием.

## Why It's Good For The Game
Обычный смертный не сможет залететь на базу шестерней через телепорт.
Копьём наконец-то можно снова бить.
Броня не будет ломаться от своего же заклинания.
Баланс

## Changelog
:cl:
add: Added EMP protection for cyborgs from ratvar act
add: if check for clockslab
add: drone language for cogscarab
tweak: how games give/return vars, so it will be easier to balance
tweak: changed damage for few weapons
tweak: speed for haste enchant
tweak: cogscarab's multitool now made from brass
fix: fixed a spear attack
fix: fixed harden plates for armor
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BeebBeebBoob committed:
Feat: Clockwork Cult (#658 #946 #952)

* Clock-rework 1

* Clock-rework 2

* Clock-Rework 3

* Clock-Rework 4

* Clock-Rework 4.1

* Clock-Rework 5

* Clock-Rework 6

* Clock-Rework 7

* Clock-Rework 8

* Vallat suggests

Co-authored-by: Vallat <vox.crit2013@gmail.com>

* Clock-Rework 9

Большое обнова предметов. Улучшение и плюшки.

* spell-wall-fix

* some dmi conflicts and mouse clockwork

* Bruh

* FUCK

* thing

* items,spellrefactor,de_shard and summonratvar

* spots fix

* REEE

* cogscarab and handies

* Gateway for ratvar e.t.c. the ritual

* ratvar can now be summoned by players

* e

* EEE

* lavaland ratvar. wat, it's not dead wtf

* robotssssssssss

* Revert "lavaland ratvar. wat, it's not dead wtf"

This reverts commit 0ce7d778bda46df0b7131fa568347bae21ec833e.

* Lavaland

* Hand spells yeaaaa

* tgui fix and cloth fix

* loop-anim-thinges

* image-conflict

* robot change

* Roboto Refactor Icons and eyes and ratvar_act

* fu

* ratvar_act and stuff

* Vallat suggests

Co-authored-by: Vallat <vox.crit2013@gmail.com>

* suggests

* more suggests

* FINAL TOUCHES

* eee

* SHIT

* return FALSE

* balance

* balance and fixes

* more fixes

* bugs

* css

* fixs and QoLs

* fix and some punishes to unclockers

* breaking upload totally

* Vallat suggests

Co-authored-by: Vallat <vox.crit2013@gmail.com>

* fixes

* smol fix

* moar fix

* fix

* FIX

* FU-

Co-authored-by: Vallat <vox.crit2013@gmail.com>

Bizzonium committed:
clockwork fix

AddRaiser committed:
Feat: add Alien Infestation back to events (#1020)

BeebBeebBoob committed:
Feat: Clockwork balancing (#1048)

* Tweak: НерфСтанов (testmerge) (#426)

* Этот патч ставит однокнопочные станы туда, где им самое место.

т.е у параши.
Большинству источников станов в игре урезано время стана до минимума и добавлен стамина урон. Теперь захват живьем можно совершить только либо через граб, либо через дисейблер, либо через продолжительный массаж станами(7 для тазеров, 10 для станбатонов)
Это изменение нивелирует станлоки как явление, за исключением стан оружия ближнего боя, которое всё еще может станлочить несмотря на то что стана не хватит на наручники)
Сильный нерф станпрода. Его стандартного зарядка не хватит для стамина крита. Только с использованием граба(или гарроты) можно захватить цель. Также подверглись изменению наручники(надеваются дольше на 50%) и часть наркотиков(дают больше регена стамины).
Кроме того усилен бакшот и ослаблены резиновые пули и дробь для дробовика.
Уменьшено время случайных станов в ближнем бою для безоружного и не очень боя.
Уменьшен стан с большинства источников подскальзывания.

* Допиливание новой боевки.

-Глейры теперь роняют на достаточное время для связывания, но любые простые общедоступные химикаты уменьшающие стан имеют шанс сократить время стана до безопасных значений(никотин, эпинефрин и прочие)
-Тазеры теперь наносят 25 стамина урона(4 попадания до стаминакрита)
-Дисйблеры наносят 33 стамина урона(те же 4 попадания до стамина крита)
=разные значения урона(при одинаковых количествах попаданий до стаминакрита) будут иначе воздействовать на цели под наркотой(регенерирующей выносливость)
-Все апгрейды грабов ускорены на 10%
-Станы от дефибов порезаны
-Станбатоны наносят на 50% больше стамина урона (стамина крит за 7 ударов
-Мут от глейра шэдоулинга возвращен к старым значениям.
-Потребление энергии дисейблером борга уменьшено с 250 до 175
-Изменены значения восстановления стамины у некоторых наркотиков и химикатов

* Update vampire_powers.dm

* Update bullets.dm

* Update energy.dm

* Update batons.dm

* Update stunbaton.dm

* Update secbot.dm

* Update secbot.dm

* Переработка глейра тенелинга

* Переработка абилок вампира

* Починка переработки глейра тенелинга

pepebrain

* Твик граба

* Твик мимовского оружия

* Твик батона

* Твик режима казни доминатора

* Твик адренала

* Твик CNS Rebooter

* Update stunbaton.dm

* Update augments_internal.dm

* Update energy.dm

Co-authored-by: Jordan <Alexansans.mail.ru>

* "Предрелизный" твик новой боёвки (#431)

* Починка рофлокода и допиливание баланса абилок

Shadowstep - На этот раз цена снижена не только в названии. 30 ->20
Jaunt - Цена повышена для баланса нового эффекта гипноза и удешевления других абилок 30 -> 50
Summon Bats - Снижена реальная цена а не только название 75 -> 50
Deseased touch - Снижена реальная цена а не только название 100 -> 50

* Update stunbaton.dm

Твик батонов
Станбатон СБ - 2 тика стана и 25 стамины
Станпрод - 1тик стана(то есть просто разоружение) и 33 стамины.

* Update code/modules/mob/living/carbon/human/human_defense.dm

Логгирование попытки

Co-authored-by: Vallat <vox.crit2013@gmail.com>

* Update code/game/gamemodes/vampire/vampire_powers.dm

фикс метровой очереди за Стамина уроном kekwhat

Co-authored-by: Vallat <vox.crit2013@gmail.com>

* Починка дефицитного стамина урона по талонам от Vallat. stamina damage for days

Co-authored-by: Vallat <vox.crit2013@gmail.com>
Co-authored-by: Jordan <Alexansans.mail.ru>

* Замена Sleep() на полноценный статус эффект (#432)

Co-authored-by: Jordan <Alexansans.mail.ru>

* БИ не пройдет (#454)

Взвинчиваем цены на БИ потому что я не забалансил их вовремя но не хочу чтобы они руинили тесты. Грязный трюк? Грязный.

Co-authored-by: Jordan <Alexansans.mail.ru>

* ClockWork Balancing and refactors

Co-authored-by: Jordanstolethepreciousslam <78968085+Jordanstolethepreciousslam@users.noreply.github.com>
Co-authored-by: Vallat <vox.crit2013@gmail.com>

AddRaiser committed:
Feat: better decompression

Fix permastun

Den109G committed:
Beebs changes

[Применение изменений Бипа и немного баланса](https://discord.com/channels/617003227182792704/981565006762106891/997387930358255716)

Den109G committed:
cogscrab drone language

Den109G committed:
soul vessel robots buff

Den109G committed:
easier for balancing + fix for armor

Den109G committed:
clockslab for ratvar

Den109G committed:
clean up

BeebBeebBoob commented on code/modules/mob/language.dm:
```diff
@@ -616,7 +616,7 @@
 
 	for(var/mob/living/S in GLOB.alive_mob_list)
-		if(drone_only && !istype(S,/mob/living/silicon/robot/drone))
+		if(drone_only && !(istype(S,/mob/living/silicon/robot/drone)||(istype(S,/mob/living/silicon/robot/cogscarab))))
```
```suggestion
		if(drone_only && !(istype(S,/mob/living/silicon/robot/drone)||iscogscarab(S)))
```

BeebBeebBoob commented on code/modules/mob/living/silicon/robot/robot.dm:
```diff
@@ -1059,10 +1059,12 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 		pick_module("Clockwork")
+		emp_protection = TRUE
+		speed = -0.5
+		pdahide = TRUE
```
`pdahide = true`
Там уже`UnlinkSelf()` убирает все признаки соединения к станции как помню

Den109G replied:
UnlinkSelf лишь скрывает мессенджер, но не выключает. Т.е. без этой строчки, боргов нельзя увидеть по мессенджеру, но можно получать от них сообщения.
Можно конечно убрать `pdahide = TRUE`, но смысла особого не вижу. Просто борги смогут писать через мессенджер (у них и так есть связь хирефанта, зачем ПДА?).
![image](https://user-images.githubusercontent.com/87372121/179738838-6d0d3ee7-d4f5-41ee-8b6e-6a85a0e657dc.png)
![image](https://user-images.githubusercontent.com/87372121/179738884-b0d9dfa9-1dc6-45c0-907a-0596245ac65a.png)
![image](https://user-images.githubusercontent.com/87372121/179739009-b721fc24-e9f9-40ca-8952-aa30c7b6cd6e.png)

Den109G replied:
(Ещё не учитываем что оно нормально не работает, ибо ПДА меняется только после смены имени)

BeebBeebBoob replied:
ну тогда вопрос отпал

BeebBeebBoob commented on code/__DEFINES/clockwork.dm:
```diff
@@ -59,31 +59,31 @@ GLOBAL_LIST_INIT(hammer_spells, list(
+	new /datum/spell_enchant("Reconstruction", RECONSTRUCT_SPELL, 500, 9)
 ))
 /// Power per crew for summoning. For example if 45 players on station, the Ratvar will demand 45*number.
 #define CLOCK_POWER_PER_CREW 400
```
```suggestion
#define CLOCK_POWER_PER_CREW 300
```
Я заметил что всегда именно задание на питание делают последним
Стоит тогда увеличить на 0.10 в 108 и 112 строке дабы сразу не раскрыли ибо в процентах все таки(хотя там палятся руки и их слегка трудно заметить но они светятся (не в темноте))

Также 92 на 50 бафнуть и 93 до 500 ибо это минус целое тело

Den109G replied:
Надеюсь правильно понял, что надо поправить.

BeebBeebBoob reviewed (changes requested):
Решить пару вопросов и балансов и можно выпустить.

Den109G committed:
cogscarab if

Co-authored-by: BeebBeebBoob <32985153+BeebBeebBoob@users.noreply.github.com>

Den109G committed:
баланс клокеров в сторону рук

меньше энергии на золотые руки, меньге энергии для конверта, больше людей для раскрытия.

Den109G committed:
Merge branch 'testmerge2' of https://github.com/Den109G/Paradise into testmerge2

BeebBeebBoob commented on code/__DEFINES/clockwork.dm:
```diff
@@ -86,11 +86,11 @@ GLOBAL_LIST_INIT(shard_spells, list(
 /// Power gains permanent
-#define CLOCK_POWER_CONVERT 150
-#define CLOCK_POWER_SACRIFICE 300
+#define CLOCK_POWER_CONVERT 100
```
92: слегка имел в виду до 200 дабы те кого законвертовали могли что-то сделать себе за это количество питания.

Den109G replied:
👌

BeebBeebBoob commented:
Последняя правка и этот ПР я считаю готовым.

Den109G committed:
правки энергии

Den109G commented:
145 файлов + конфликты... Будет весело.

BeebBeebBoob commented:
А как так получилось??

Den109G commented:
После force-push'а появились. Ну ничего, мне не лень пересоздавать ПР.

Den109G commented:
А говорил я себе. Не спи. Они снова нападут на ПР.
Ну не поверил я себе, вот что случилось. 🤡
